homeassistant/components/homematic/__init__.py

```diff
@@ -139,6 +139,7 @@
                     cv.match_all: {
                         vol.Required(CONF_HOST): cv.string,
                         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                        vol.Optional(CONF_JSONPORT, default=DEFAULT_JSONPORT): cv.port,
```

Comment:
This is a new option for the hosts section in the config schema. It's not mentioned in the documentation for the hosts section either only under interfaces.

Why should this be considered a fix?

Generally, we no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

Reply:
Well, the port was always hardcoded, but should have been configurable from the beginning (and it was in the pyhomematic dependency, the integration just didn't include it). In most cases the hardcoded value works, but for some setups an alternate port is required. This usually is the case when people introduce SSL, which wasn't possible with the old version of the HomeMatic hub.

So essentially not allowing a different port would force the users to not use SSL at all (within the HomeMatic environment) if they require the host-entity.

Reply:
And yes, this would also require an update of the documentation.

Reply:
@pvizeli what do you think?

Reply:
I just updated the configuration, see https://github.com/home-assistant/home-assistant.io/pull/16935
Hopefully this small fix will get into the master branch :)

Reply:
I also have trouble using homematic with ssl. This change would fix the issue.

Reply:
I agree with martin. Using ADR10 would also be a huge benefit for HM

Reply:
@pvizeli @martinhjelmare I really do respect your rule regarding ADR10, but its really a pitty when not accepting such a small fix for different problems.  This desicion will lead to frustrated users (like me) and lower security withour SSL. You should consider how much work it would be to change the integration completely to use the config flow compared to this two-line-fix.

Reply:
@pvizeli @MartinHjelmare I agree totally with @nicx. The fix will not harm anyone or the quality of the project and will not frustrate users and contributes, which was the aim of ADR10. Changing the integration of HM should be a long term goal but you should not stop HM from working correctly if the solution to get it work is such a small thing.

Reply:
@pvizeli @MartinHjelmare any chance to get this into master? :)

Reply:
I don't think we should do it and Pascal seems to agree. I think this integration should be reworked to use config flow and config entries first.

Reply:
@MartinHjelmare You are preventing home assistant from working correctly because you rely on something that should prevent users and contributors against frustration and with that you are frustrate the users and the contributors. To be honest this is ridiculous, even since @danielperna84 said the port was hardcoded and the only thing this PR is change is making this configurable.

Reply:
Yes and we do not allow configuration changes, please respect the architectural decision made by the project.